stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -69,7 +70,8 @@ internal class ChatDomainImpl internal constructor(internal val chatDomainStateF
             chatDomainStateFlow.currentUser = value
         }
 
-    override val user: LiveData<User?> = chatDomainStateFlow.user.asLiveData()
+    override val user: LiveData<User?>
+        get() = (chatDomainStateFlow as ChatDomainImpl)._userLiveData
```

Comment:
asLiveData doesn't pick the `.value` in the `StateFlow` so this value gets lost in the convertion and only the posted values get picked. 

That becomes a problem in the `ChannelListViewModel` as I don't see a way to wait to the currentUser.id as the filter needs it in the constructor. 

Only simple option that I see is to use the `offline.ChatDomain` (StateFlow) instead of `livedata.ChatDomain` inside `ChannelListViewModel`. That would change the API, but it seams a change that is going to happen anyways. What do you think about it?

Reply:
I applied the new ChatDomain, then it is possible to remove this line.